apps/src/applab/applab.js

```diff
@@ -1306,6 +1309,17 @@ Applab.onPropertyChange = function(element, name, value) {
       element.style.backgroundImage = 'url(' + value + ')';
       element.style.backgroundSize = width + 'px ' + height + 'px';
       break;
+    case 'picture':
+      element.src = value;
+      element.onload = function () {
+        element.style.width = this.naturalWidth + 'px';
```

Comment:
Agreed on both comments. Using this does feel a little strange here. I think what I might actually do is replace this with element, and add a comment explaining that naturalWidth/naturalHeight aren't populated until onload is called.